tests/testthat/test-factor-level-coercion.R

```diff
@@ -0,0 +1,926 @@
+  
+  # Check that the predictions are the same
+  expect_equal(rfOutDf1[1, ]$PredictedProbNBR, rfOutDf2[1, ]$PredictedProbNBR)
+})
+
+test_that("Lasso classification predictions are independent of each other", {
+  # Single row prediction data set
+  dfDeploy1 <- data.frame(id = 9001,
+                          length = 6,
+                          diameter = 3,
+                          heat = "Cold",
+                          condiment = "Syrup")
+  
+  p1 <- SupervisedModelDeploymentParams$new()
+  p1$type <- "classification"
+  p1$df <- dfDeploy1
+  p1$grainCol <- "id"
+  p1$predictedCol <- "isHotDog"
+  p1$impute <- TRUE
+  p1$debug <- F
+  p1$cores <- 1
+  
+  # Two row prediction data set, with first row same as dfDeploy2
+  dfDeploy2 <- data.frame(id = c(9001, 9002),
+                          length = c(6, 2),
+                          diameter = c(3, 2), 
+                          heat = c("Cold", "Hot"),
+                          condiment = c("Syrup", "Mustard"))
+  
+  p2 <- SupervisedModelDeploymentParams$new()
+  p2$type <- "classification"
+  p2$df <- dfDeploy2
+  p2$grainCol <- "id"
+  p2$predictedCol <- "isHotDog"
+  p2$impute <- TRUE
+  p2$debug <- F
+  p2$cores <- 1
+  
+  # Deploy lasso on single row
+  capture.output(lassoD1 <- LassoDeployment$new(p1))
+  lassoD1$deploy()
+  
+  lassoOutDf1 <- lassoD1$getOutDf()
+  
+  # Deploy lasso on 2 rows, one of which is identical to the one above
+  capture.output(lassoD2 <- LassoDeployment$new(p2))
+  lassoD2$deploy()
+  
+  lassoOutDf2 <- lassoD2$getOutDf()
+  
+  # Check that the predictions are the same
+  expect_equal(lassoOutDf1[1, ]$PredictedProbNBR, lassoOutDf2[1, ]$PredictedProbNBR)
+})
+
+test_that("Extra factors are imputed correctly for rf classification (1 column)", {
+  # Data set with new factor level in one column
+  # Need last row for imputation
+  dfDeploy3 <- data.frame(id = c(9003, 9004, 9005),
+                          length = c(6, 6, 6),
+                          diameter = c(3, 3, 3), 
+                          heat = c("Frozen", NA, "Cold"),
+                          condiment = c("Ketchup", "Ketchup", "Ketchup"))
+  
+  p3 <- SupervisedModelDeploymentParams$new()
+  p3$type <- "classification"
+  p3$df <- dfDeploy3
+  p3$grainCol <- "id"
+  p3$predictedCol <- "isHotDog"
+  p3$impute <- TRUE
+  p3$debug <- F
+  p3$cores <- 1
+  
+  # Deploy on 3 rows, which should yield the same predictions
+  capture.output(rfD3 <- RandomForestDeployment$new(p3))
+  
+  # Check that a warning reporting new factor levels is triggered
+  expect_warning(rfD3$deploy(), 
+                 paste("New categorical variable levels were found:\n",
+                       " -  heat : Frozen\n",
+                       "These values have been set to NA.",
+                       sep = ""))
+  
+  rfOutDf3 <- rfD3$getOutDf()
+  
+  # Check that new value is treated as NA
+  expect_equal(rfOutDf3[1, ]$PredictedProbNBR, rfOutDf3[2, ]$PredictedProbNBR)
+})
```

Comment:
No, this just checks that the new factor level is treated in the same way as NA.  

However, there is a separate unit test file for imputation (test-impute-column).